providers/implementations/kdfs/krb5kdf.c

```diff
@@ -359,7 +359,7 @@ static int cipher_init(EVP_CIPHER_CTX *ctx,
     klen = EVP_CIPHER_CTX_get_key_length(ctx);
     if (key_len != (size_t)klen) {
         ret = EVP_CIPHER_CTX_set_key_length(ctx, key_len);
-        if (!ret)
```

Comment:
this is not quite right still.. you may want to do set ret = 0 ; The calls to cipher_init in this file are doing !ret checks after the call also.

Reply:
Thanks for the review, you are right. Changed already.

Reply:
You seem to be doing a good job of fixing all these returns.. Can you check your other PR's for this kind of problem. This might be the only one.